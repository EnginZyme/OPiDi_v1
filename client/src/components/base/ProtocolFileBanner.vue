<template>
  <div class="protocol-file-banner">
    <div class="protocol-file-banner__file-name">{{ cachedProtocolName }}</div>
    <div class="protocol-file-banner__author-name"> {{ cachedProtocolAuthor }}</div>
  </div>
</template>

<script>
import { mapGetters } from "vuex";

/**
 * A component that renders and styles the name of the protocol and the name of the author as well.
 */
export default {
  name: "ProtocolFileBanner",
  computed: {
    ...mapGetters(["cachedProtocolAuthor", "cachedProtocolName"]),
  },
};
</script>

<style scoped>
@import url('https://fonts.googleapis.com/css?family=Cedarville+Cursive');

.protocol-file-banner {
    display: grid;
    width: 99%;
    grid-template-columns: 2fr auto;
    grid-column-gap: 10px;
    align-items: center;
}
.protocol-file-banner__file-name {
  color: #ffffff;
  font-size: 20px;
  font-weight: 500;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
.protocol-file-banner__author-name {
  color: #ffffff;
  font-size: 14px;
  text-align: end;
  font-family: 'Cedarville Cursive', cursive;;
}
</style>
