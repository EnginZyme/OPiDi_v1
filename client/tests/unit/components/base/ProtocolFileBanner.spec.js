import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import ProtocolFileBanner from "@/components/base/ProtocolFileBanner";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("shallowMounted ProtocolFileBanner", () => {
  let getters;
  let store;
  let wrapper;

  const sampleProtocolName = "sample-protocol-name";
  const sampleProtocolAuthorName = "sample-protocol-author-name";

  beforeEach(() => {
    getters = {
      cachedProtocolName: () => sampleProtocolName,
      cachedProtocolAuthor: () => sampleProtocolAuthorName,
    };

    store = new Vuex.Store({
      getters,
    });

    wrapper = shallowMount(ProtocolFileBanner, {
      store,
      localVue,
    });
  });

  it("contains sample protocol name", () => {
    expect(wrapper.text()).toContain(sampleProtocolName);
  });

  it("contains the sample protocol author name", () => {
    expect(wrapper.text()).toContain(sampleProtocolAuthorName);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
