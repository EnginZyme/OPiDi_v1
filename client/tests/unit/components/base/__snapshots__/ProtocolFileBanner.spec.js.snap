// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted ProtocolFileBanner renders correctly 1`] = `
<div
  class="protocol-file-banner"
>
  <div
    class="protocol-file-banner__file-name"
  >
    sample-protocol-name
  </div>
   
  <div
    class="protocol-file-banner__author-name"
  >
     sample-protocol-author-name
  </div>
</div>
`;
